codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/util/Smithy.kt

```diff
@@ -98,8 +98,8 @@ fun Shape.shouldRedact(model: Model): Boolean =
 
 const val REDACTION = "\"*** Sensitive Data Redacted ***\""
 
-fun Shape.redactIfNecessary(model: Model, safeToPrint: String): String =
-    if (this.shouldRedact(model)) {
+fun Shape.redactIfNecessary(model: Model, safeToPrint: String, sensitiveMember: Boolean = false): String =
+    if (sensitiveMember || this.shouldRedact(model)) {
```

Comment:
Shouldn't this `sensitiveMember` check be covered by the `shouldRedact` function? Probably checking `shouldRedact` on its `container`?